### PR TITLE
Fix mapping rest parameters when inlining a function with an inner inlinee

### PR DESCRIPTION
**Pull request by meg-gupta**
While mapping formals during inlining, ArgIn_Rest was handled incorrectly when it belonged to
another inlined function originally in the inlinee. For such ArgIn_Rest instructions we should generate code depending on the formalCount of ArgIn_Rest's function instead of the formalCount of the current inlinee.

### CODE REVIEW

**Comment by meg-gupta** [comment]
@rajatd and @ThomsonTan Can you please review ?

https://github.com/Microsoft/ChakraCore/pull/2202 was the PR that resulted into this problem.

**Comment by meg-gupta** [comment]
Looking at the test failure.

**Comment by ThomsonTan** [comment]
Also is it possible to add a test case to expose this?

File: lib/Backend/Inline.cpp [hunk 1]
```diff
@@ -4677,8 +4677,7 @@ Inline::MapFormals(Func *inlinee,
             }
             if (instr->m_func != inlinee)
             {
-                // this can happen only when we are inlining a function which has inlined an apply call with the arguments object
-                formalCount = instr->m_func->GetJITFunctionBody()->GetInParamsCount();
```
Comment (satheeshravi) [inline lib/Backend/Inline.cpp]: >formalCount [](start = 16, length = 11)

So the earlier fix about formalCount being wrong during NewScArray creation - is also taken care? - can you also pls  verify that?

Comment (meg-gupta) [inline lib/Backend/Inline.cpp]: Yes, the updated commit handles correct codegen for ArgIn_Rest.

**Comment by rajatd** [comment]
:shipit:

**Pull request merged**